tests/e2e/unified_test.go

```diff
@@ -273,11 +274,42 @@ func testFixturesInSeries(ctx context.Context, t *testing.T, testPause bool, can
 				}
 				create.RunCreateDeleteTest(h, opt)
 
-				for _, obj := range exportResources {
-					got := exportResource(h, obj)
-					if got != "" {
-						expectedPath := filepath.Join(fixture.SourceDir, "export.yaml")
-						h.CompareGoldenFile(expectedPath, string(got), IgnoreComments, ReplaceString(project.ProjectID, "example-project-id"))
+				if os.Getenv("GOLDEN_OBJECT_CHECKS") != "" {
+					for _, obj := range exportResources {
+						// Construct file path to save golden files
+						dir := "golden/"
+						pieces := strings.Split(t.Name(), "/")
+						var testName string
+						if len(pieces) > 0 {
+							testName = pieces[len(pieces)-1]
+						} else {
+							t.Errorf("failed to get test name")
+						}
+						// Golden test exported GCP object
+						got := exportResource(h, obj)
+						if got != "" {
+							expectedPath := filepath.Join(dir, fmt.Sprintf("export/_%v.yaml", testName))
+							h.CompareGoldenFile(expectedPath, string(got), IgnoreComments, ReplaceString(project.ProjectID, "example-project-id"))
+						}
+						// Golden test KRM object schema
+						u, err := getKRMObject(h, obj)
+						if u == nil {
+							t.Errorf("failed to get KRM object: %v", err)
+						} else {
+							if err := normalizeObject(u, project, uniqueID); err != nil {
+								t.Fatalf("error from normalizeObject: %v", err)
+							}
+							got, err := yaml.Marshal(u)
+							if err != nil {
+								t.Errorf("failed to convert KRM object to yaml: %v", err)
+							}
+							// File name looks like _testname.yaml
+							expectedPath := filepath.Join(dir, fmt.Sprintf("object/_%v.yaml", testName))
+							err = test.CompareGoldenObject(expectedPath, got)
```

Comment:
Nit: personally I think it's fine to pass `t` in: it makes it clear this is a test method, and it means we can e.g. call t.Errorf or t.Fatalf

Reply:
As this is the second PR I reviewed for unified_test.go, I'm still getting used to the coding style here.

IMO `CompareGoldenFile()` or `ComputeGoldenObject()` don't look like a test util function due to their vague names. It's reasonable or even preferred that we pass in `t` in a test util function, but I think the name should be more specific and test-related, e.g. `AssertRetrievedYAMLMatchesGoldenFile`, `AssertKRMObjectMatchesGoldenObject`.

Reply:
As mentioned in [here](https://github.com/GoogleCloudPlatform/k8s-config-connector/pull/1447#discussion_r1550645435), I think it's easier to address Joyce's comment in another task when we do a refactor of unified_test. I created a ticket b/332777103.